src/main/java/com/google/sps/servlets/ChatServlet.java

```diff
@@ -0,0 +1,64 @@
+package com.google.sps.servlets;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.StructuredQuery.OrderBy;
+import com.google.gson.Gson;
+import main.java.com.google.sps.data.Chat;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+@WebServlet("/chats")
+public class ChatServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // OBJECTIVE: Create a chat between users
+    // Step 1: Get the user and the partner ids
+    String participant1 = Jsoup.clean(request.getParameter("userid"), Whitelist.none());
+    String participant2 = Jsoup.clean(request.getParameter("userid"), Whitelist.none());
```

Comment:
Hmm, not the way I'd recommend getting a specific chat (which would use the chat `id`) or a list of chats for a specific user (which uses just a single `userId`)